src/blob/index/tools.rs

```diff
@@ -194,15 +11,30 @@ pub(crate) fn get_hash(buf: &[u8]) -> Vec<u8> {
 
 // if there is no elements, data will be wrong (because we can't get key_size),
 // BUT it will be computed correctly during first push
-pub(crate) fn compute_mem_attrs(record_headers: &InMemoryIndex) -> MemoryAttrs {
+pub(crate) fn compute_mem_attrs(
+    record_headers: &InMemoryIndex,
+    records_count: usize,
+) -> MemoryAttrs {
     let key_size = record_headers.keys().next().map_or_else(|| 0, |v| v.len());
```

Comment:
I think you should return Err or panic if `key.len() == 0`

Reply:
I'll panic